### DISCUSSION TITLE
Token Consumption for  Embedding API

### DISCUSSION THREAD

**Discussion by jmurkothintel** [microsoft/semantic-kernel#6592] [Q&A] [answered]
In semantic kernel implementation of the  GetEmbeddingAsync Method, though the api returns the usage information, that is not bubbled up . Is there any reason why it is not? Also is there any plans to bubble  it up to the call chain and make  available to the caller?
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs#L320

**Comment by dmytrostruk** [comment, marked as answer]
@jmurkothintel Thanks for your question. 
Current embeddings API doesn't allow to include metadata, since it returns multiple `ReadOnlyMemory` objects only:
https://github.com/microsoft/semantic-kernel/blob/bebe12720e5e230061709bf89aa093a3d2b2c6b9/dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/IEmbeddingGenerationService.cs#L28

We have ADR for new embedding design, which takes that into account and allows to return metadata as well. We will need to prioritize this item.

**Reply by jmurkothintel** [reply]
@dmytrostruk Was just following up to see if the design is complete and this request has been prioritized

**Reply by dmytrostruk** [reply]
@jmurkothintel We are currently focused on memory-related functionality and connectors, which should include re-design for embeddings API.
